File: apps/blog/src/app/application/services/post.write.service.ts
import { injectable, inject } from 'inversify';
import { Post } from '../../domain/entities/post';
import type { PostRepository } from '../../domain/repositories/post.repository';

@injectable()
export class PostWriteService {
  constructor(
    @inject('PostRepository') private postRepository: PostRepository
  ) {}

  async createPost(post: Post): Promise<void> {
    await this.postRepository.save(post);
  }
}

File: back/lib/blog/application/post.write_service.dart
import 'package:staretz_domain/blog/domain/entities/post.dart';
import 'package:staretz_domain/blog/domain/ports/post_repository.dart';
import 'package:staretz_domain/blog/domain/value_objects/post_slug.dart';

class PostWriteService {
  final PostRepository _repository;

  PostWriteService(this._repository);

  Future<void> save(Post post) => _repository.save(post);

  Future<void> update(PostSlug originalSlug, Post post) =>
      _repository.update(originalSlug, post);

  Future<void> delete(PostSlug slug) => _repository.delete(slug);
}
